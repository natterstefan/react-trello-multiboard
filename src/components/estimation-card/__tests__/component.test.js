import React from 'react'
import { shallow } from 'enzyme'
import { assign } from 'lodash'

import LinearProgress from '@material-ui/core/LinearProgress'
import EstimationCard from '../component'

const props = {
  error: null,
  boards: {
    data: [
      {
        board: {
          id: 'b1',
          name: 'Board1',
        },
        estimations: {
          consumed: 1,
          estimated: 2,
        },
      },
    ],
  },
  isLoading: false,
  members: {
    members: [{ id: 'member-1' }],
    'member-1': {
      boardEstimations: {
        b1: {
          consumed: 3,
          estimated: 3,
        },
      },
    },
  },
  memberToggle: {
    togglePreferred: false,
    togglePreferredMember: null,
  },
}

describe('Component/EstimationCard', () => {
  test('should render without throwing an error', () => {
    expect(shallow(<EstimationCard {...props} />)).toMatchSnapshot()
  })

  test('should render an LinearProgress when component is waiting for loading data to finish', () => {
    const wrapper = shallow(<EstimationCard {...props} isLoading />)
    expect(wrapper.find(LinearProgress).length).toEqual(1)
  })

  test('should render an empty span when an error occured', () => {
    const wrapper = shallow(<EstimationCard {...props} error="sample error" />)
    expect(wrapper.html()).toEqual('<span></span>')
  })

  test('should render an empty span when the boards prop has no data array', () => {
    const newProps = assign({}, props, {
      boards: {
        data: [],
      },
    })
    const wrapper = shallow(<EstimationCard {...newProps} />)
    expect(wrapper.html()).toEqual('<span></span>')
  })

  test('should render estimations of a toggled member', () => {
    const memberToggle = {
      togglePreferred: true,
      togglePreferredMember: 'member-1',
    }

    const wrapper = shallow(<EstimationCard {...props} memberToggle={memberToggle} />)
    expect(wrapper.find('strong').text()).toEqual('Total: (3)[3]')
  })

  test('should render fallback estimations if the toggled member is not valid', () => {
    const memberToggle = {
      togglePreferred: true,
      togglePreferredMember: 'member-2',
    }

    const wrapper = shallow(<EstimationCard {...props} memberToggle={memberToggle} />)
    expect(wrapper.find('strong').text()).toEqual('Total: (0)[0]')
  })
})
