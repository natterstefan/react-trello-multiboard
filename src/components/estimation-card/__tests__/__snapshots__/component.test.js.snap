// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component/EstimationCard should render without throwing an error 1`] = `
<div
  style={
    Object {
      "display": "flex",
    }
  }
>
  <WithStyles(Card)>
    <WithStyles(CardContent)>
      <WithStyles(List)>
        <WithStyles(ListItem)
          button={true}
          dense={true}
          key="b1"
        >
          <WithStyles(Typography)>
            Board1
            : (
            2
            )[
            1
            ]
          </WithStyles(Typography)>
        </WithStyles(ListItem)>
        <WithStyles(ListItem)
          button={true}
          dense={true}
          key="estimations-2"
        >
          <WithStyles(Typography)>
            <strong>
              Total: (
              2
              )[
              1
              ]
            </strong>
          </WithStyles(Typography)>
        </WithStyles(ListItem)>
      </WithStyles(List)>
    </WithStyles(CardContent)>
  </WithStyles(Card)>
</div>
`;
